streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -166,4 +166,14 @@ public void shouldListAllTaskDirectories() throws Exception {
         assertTrue(dirs.contains(taskDir2));
     }
 
+    @Test
+    public void shouldCreateDirectoriesIfParentDoesntExist() throws Exception {
+        final File tempDir = TestUtils.tempDirectory();
+        final File stateDir = new File(new File(tempDir, "foo"), "state-dir");
```

Comment:
I think the implicit assumption is that the parent "foo" doesn't exist (yet), right?  If so, shouldn't we assert that it actually does not exist prior to continuing?